lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1118,6 +1121,38 @@ void LLVMToSPIRVBase::transFPGAFunctionMetadata(SPIRVFunction *BF,
     transMetadataDecorations(FDecoMD, BF);
 }
 
+void LLVMToSPIRVBase::transFunctionMetadataAsExecutionMode(SPIRVFunction *BF,
+                                                           Function *F) {
+  if (!BM->isAllowedToUseExtension(ExtensionID::SPV_INTEL_maximum_registers))
```

Comment:
Do we not expect this function to be reused for other extensions like this in the future? This check seems redundant.

Thanks

Reply:
Indeed redundant, will remove